components/omega/src/ocn/Eos.h

```diff
@@ -0,0 +1,296 @@
+      const Real V231 = 3.1655306078e-07;
+      const Real V041 = -3.4102187482e-06;
+      const Real V141 = 1.2956717783e-06;
+      const Real V051 = 5.0736766814e-07;
+      const Real V002 = 3.0623833435e-06;
+      const Real V102 = -5.8484432984e-07;
+      const Real V202 = -4.8122251597e-06;
+      const Real V302 = 4.9263106998e-06;
+      const Real V402 = -1.7811974727e-06;
+      const Real V012 = -1.1736386731e-06;
+      const Real V112 = -5.5699154557e-06;
+      const Real V212 = 5.4620748834e-06;
+      const Real V312 = -1.3544185627e-06;
+      const Real V022 = 2.1305028740e-06;
+      const Real V122 = 3.9137387080e-07;
+      const Real V222 = -6.5731104067e-07;
+      const Real V032 = -4.6132540037e-07;
+      const Real V132 = 7.7618888092e-09;
+      const Real V042 = -6.3352916514e-08;
+      const Real V003 = -3.8088938393e-07;
+      const Real V103 = 3.6310188515e-07;
+      const Real V203 = 1.6746303780e-08;
+      const Real V013 = -3.6527006553e-07;
+      const Real V113 = -2.7295696237e-07;
+      const Real V023 = 2.8695905159e-07;
+      const Real V004 = 8.8302421514e-08;
+      const Real V104 = -1.1147125423e-07;
+      const Real V014 = 3.1454099902e-07;
+      const Real V005 = 4.2369007180e-09;
+
+      SpecVolPCoeffs(5, K) = V005;
+      SpecVolPCoeffs(4, K) = V014 * Tt + V104 * Ss + V004;
+      SpecVolPCoeffs(3, K) = 
+            (V023 * Tt + V113 * Ss + V013) * Tt + 
+            (V203 * Ss + V103) * Ss + V003;
+      SpecVolPCoeffs(2, K) = 
+            (((V042 * Tt + V132 * Ss + V032) * Tt + 
+            (V222 * Ss + V122) * Ss + V022) * Tt +
+            ((V312 * Ss + V212) * Ss + V112) * Ss + V012) * Tt +
+            (((V402 * Ss + V302) * Ss + V202) * Ss + V102) * Ss + V002;
+      SpecVolPCoeffs(1, K) = 
+            ((((V051 * Tt + V141 * Ss + V041) * Tt + 
+            (V231 * Ss + V131) * Ss + V031) * Tt +
+            ((V321 * Ss + V221) * Ss + V121) * Ss + V021) * Tt +
+            (((V411 * Ss + V311) * Ss + V211) * Ss + V111) * Ss + V011) * Tt +
+            ((((V501 * Ss + V401) * Ss + V301) * Ss + V201) * Ss + V101) * Ss + V001;
+      SpecVolPCoeffs(0, K) =
+            (((((V060 * Tt + V150 * Ss + V050) * Tt + 
+            (V240 * Ss + V140) * Ss + V040) * Tt +
+            ((V330 * Ss + V230) * Ss + V130) * Ss + V030) * Tt +
+            (((V420 * Ss + V320) * Ss + V220) * Ss + V120) * Ss + V020) * Tt +
+            ((((V510 * Ss + V410) * Ss + V310) * Ss + V210) * Ss + V110) * Ss + V010) * Tt +
+            (((((V600 * Ss + V500) * Ss + V400) * Ss + V300) * Ss + V200) * Ss + V100) * Ss + V000;
+
+      // could insert a check here (abs(value)> 0 value or <e+33)
+   }
+
+   /// Evaluate pressure polynomial delta for TEOS-10
+   KOKKOS_FUNCTION Real calcDelta(const Array2DReal &SpecVolPCoeffs, 
+                                  const I4 K, const Real P) const {
+   
+      // Check if the pressure coefficients have been initialized
+      if (!PCoeffsInit) {
```

Comment:
The return statement causes an error with cray(or hip) compiler on Frontier because it returns nothing, even though the calcDelta function's return type is Real. I suggest moving the if statement outside the kernel function to both fix the error and prevent all GPU threads from executing that part of the code.